src/blockchain/alt_block_tree.cpp

```diff
@@ -173,13 +172,10 @@ void AltBlockTree::setPayloads(index_t& index, const PopData& payloads) {
                payloads.atvs.size(),
                index.toShortPrettyString());
 
-  VBK_ASSERT_MSG(!index.hasFlags(BLOCK_HAS_PAYLOADS),
+  VBK_ASSERT_MSG(index.getValidityLevel() == BLOCK_VALID_TREE,
                  "block %s already contains payloads",
```

Comment:
Assert message is wrong

Reply:
it may not be technically correct as in theory it could be called on a block with BLOCK_VALID_UNKNOWN status due to a bug, but in all other cases setPayloads() is the function that raises the validity level from  BLOCK_VALID_TREE to BLOCK_HAS_PAYLOADS